### PR TITLE
Add SecretManager integration support to GCF.

### PR DESCRIPTION
**Pull request by agarg2008**
Add SecretManager integration support to GCF.

https://github.com/hashicorp/terraform-provider-google/issues/9946

If this PR is for Terraform, I acknowledge that I have:

- [X] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [X] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-downstream-tools), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [X] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/third_party/terraform/tests) (for handwritten resources or update tests).
- [X] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/master/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [X] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/master/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

```release-note:enhancement
gcf: Added SecretManager integration support to GCF.
```

### CODE REVIEW

File: mmv1/third_party/terraform/website/docs/r/cloudfunctions_function.html.markdown [hunk 1]
```diff
@@ -175,6 +179,32 @@ which to observe events. For example, `"myBucket"` or `"projects/my-project/topi
...
+
+* `secret` - (Required) ID of the secret in secret manager (not the full resource name).
+
+* `version` - (Required) Version of the secret (version number or the string "latest"). It is recommended to use a numeric version for secret environment variables as any updates to the secret value is not reflected until new clones start.
+
+<a name="nested_secret_volumes"></a>The `secret_volumes` block supports:
+
+* `mount_path` - (Required) The path within the container to mount the secret volume. For example, setting the mount_path as "/etc/secrets" would mount the secret value files under the "/etc/secrets" directory. This directory will also be completely shadowed and unavailable to mount any other secrets. Recommended mount paths: "/etc/secrets" Restricted mount paths: "/cloudsql", "/dev/log", "/pod", "/proc", "/var/log".
+
+* `project_id` - (Required) Project identifier (preferably project number but can also be the project ID) of the project that contains the secret. If not set, it will be populated with the function's project assuming that the secret exists in the same project as of the function.
```
[the remark this answers is not in the document]
Comment (agarg2008) [reply mmv1/third_party/terraform/website/docs/r/cloudfunctions_function.html.markdown]: Done.

File: mmv1/third_party/terraform/resources/resource_cloudfunctions_function.go [hunk 2]
```diff
@@ -315,6 +315,80 @@ func resourceCloudFunctionsFunction() *schema.Resource {
...
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"key": {
+							Type:        schema.TypeString,
+							Required:    true,
+							Description: `Name of the environment variable.`,
+						},
+						"project_id": {
+							Type:        schema.TypeString,
+							Computed:    true,
```
[the remark this answers is not in the document]
Comment (agarg2008) [reply mmv1/third_party/terraform/resources/resource_cloudfunctions_function.go]: Done.

File: mmv1/third_party/terraform/tests/resource_cloudfunctions_function_test.go.erb [hunk 3]
```diff
@@ -918,3 +1002,152 @@ resource "google_cloudfunctions_function" "function" {
...
+data "google_project" "project" {}
+
+resource "google_project_iam_member" "gcfadmin" {
+  project = data.google_project.project.project_id
+  role     = "roles/editor"
+  member   = "serviceAccount:service-%s@gcf-admin-robot.iam.gserviceaccount.com"
+}
+
+resource "google_service_account" "cloud_function_runner" {
+  account_id   = "cloud-function-service"
```
Comment (slevenick) [inline mmv1/third_party/terraform/tests/resource_cloudfunctions_function_test.go.erb:1017]: Can we use a `tf-test-%{random_suffix}` name here? Tests are failing due to repeated names

File: mmv1/third_party/terraform/tests/resource_cloudfunctions_function_test.go.erb [hunk 4]
```diff
@@ -918,3 +1002,152 @@ resource "google_cloudfunctions_function" "function" {
 }
 `, projectNumber, networkName, vpcConnectorName, vpcConnectorName, vpcIp, bucketName, zipFilePath, functionName, vpcConnectorName)
 }
+
+func testAccCloudFunctionsFunction_secretEnvVar(projectNumber, secretName, versionName, bucketName, functionName, versionNumber, zipFilePath string) string {
+	return fmt.Sprintf(`
+data "google_project" "project" {}
+
+resource "google_project_iam_member" "gcfadmin" {
```
Comment (slevenick) [inline mmv1/third_party/terraform/tests/resource_cloudfunctions_function_test.go.erb:1010]: Why do we need to grant the auto generated service account permissions specifically here?

**Review by slevenick — CHANGES_REQUESTED** [review]

File: mmv1/third_party/terraform/tests/resource_cloudfunctions_function_test.go.erb (hunk 3 above)
Comment (agarg2008) [reply mmv1/third_party/terraform/tests/resource_cloudfunctions_function_test.go.erb]: Done.

File: mmv1/third_party/terraform/tests/resource_cloudfunctions_function_test.go.erb (hunk 4 above)
Comment (agarg2008) [reply mmv1/third_party/terraform/tests/resource_cloudfunctions_function_test.go.erb]: must have been from another test setup. deleted it and test still works :)

**Review by slevenick — APPROVED** [review]
Test failures unrelated, CloudFunctions tests look good, thanks!

**Pull request merged**